### ISSUE TITLE
supertest 1.1.0 Uncaught TypeError: Object.keys called on non-object 

### ISSUE THREAD

**Issue by ghinks** [rprieto/supersamples#26]
Firstly thank you very much for this wonderful package, I just gave a talk at a meetup in New York and I presented supersamples as one of items in the talk. 

I have just updated my node_modules and I noticed this error when the supertest package was updated from 1.0.1 to 1.1.0

[ fork with bug https://github.com/ghinks/supersamples.git](https://github.com/ghinks/supersamples.git)

I forked supersamples and update the package.json updating supertest from "~0.13.0" to "1.1.0" and I saw the following 
```
Glenn-Hinkss-MacBook:supersamples developer$ mocha
Instrumenting  superagent
Instrumenting  supertest


  instrument supertest / superagent
    request
      1) inspects the method and path
      2) inspects custom headers
      3) ignores headers with no value, or from the blacklist
      4) inspects JSON payloads
      5) inspects binary payloads
    response
      6) inspects the response status and body
      7) can inspect JSON bodies as well
      8) can inspect responses with no bodies
      9) extracts headers that were asserted on

  instruments / restify
    route
      ✓ should create route without params
      ✓ should create route with params
      ✓ should return null if route does not exists
    params
      ✓ should tokenise params

  duplicate-names linter
    ✓ should not have any errors if there are no duplicate names
    ✓ should not have an error for each duplicate name

  multiline-names linter
    ✓ should not have any errors if there are no samples with names with multiple lines
    ✓ should have an error for each multiline name

  curl
    GET
      ✓ to a given URL
      ✓ with headers
    POST
      ✓ with a JSON payload
      ✓ with a binary payload

  sample
    10) uses the <it> as a default sample name
    11) 
  removes surrounding whitespace from a samples name  

    12) can override the name with a custom property

  custom highlight.js languages
    ✓ is valid


  13 passing (68ms)
  12 failing

  1) instrument supertest / superagent request inspects the method and path:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at responseHeaders (lib/instruments/supertest.js:57:22)
      at extractResponse (lib/instruments/supertest.js:43:14)
      at Test.exports.assert (lib/instruments/supertest.js:15:15)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:156:55)
      at Test.proxy [as assert] (node_modules/sinon/lib/sinon/spy.js:82:26)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1419:10

  2) instrument supertest / superagent request inspects custom headers:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at responseHeaders (lib/instruments/supertest.js:57:22)
      at extractResponse (lib/instruments/supertest.js:43:14)
      at Test.exports.assert (lib/instruments/supertest.js:15:15)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:156:55)
      at Test.proxy [as assert] (node_modules/sinon/lib/sinon/spy.js:82:26)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1419:10

  3) instrument supertest / superagent request ignores headers with no value, or from the blacklist:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at responseHeaders (lib/instruments/supertest.js:57:22)
      at extractResponse (lib/instruments/supertest.js:43:14)
      at Test.exports.assert (lib/instruments/supertest.js:15:15)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:156:55)
      at Test.proxy [as assert] (node_modules/sinon/lib/sinon/spy.js:82:26)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1419:10

  4) instrument supertest / superagent request inspects JSON payloads:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at responseHeaders (lib/instruments/supertest.js:57:22)
      at extractResponse (lib/instruments/supertest.js:43:14)
      at Test.exports.assert (lib/instruments/supertest.js:15:15)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:156:55)
      at Test.proxy [as assert] (node_modules/sinon/lib/sinon/spy.js:82:26)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1419:10

  5) instrument supertest / superagent request inspects binary payloads:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at responseHeaders (lib/instruments/supertest.js:57:22)
      at extractResponse (lib/instruments/supertest.js:43:14)
      at Test.exports.assert (lib/instruments/supertest.js:15:15)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:156:55)
      at Test.proxy [as assert] (node_modules/sinon/lib/sinon/spy.js:82:26)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1419:10

  6) instrument supertest / superagent response inspects the response status and body:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at responseHeaders (lib/instruments/supertest.js:57:22)
      at extractResponse (lib/instruments/supertest.js:43:14)
      at Test.exports.assert (lib/instruments/supertest.js:15:15)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:156:55)
      at Test.proxy [as assert] (node_modules/sinon/lib/sinon/spy.js:82:26)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1419:10

  7) instrument supertest / superagent response can inspect JSON bodies as well:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at responseHeaders (lib/instruments/supertest.js:57:22)
      at extractResponse (lib/instruments/supertest.js:43:14)
      at Test.exports.assert (lib/instruments/supertest.js:15:15)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:156:55)
      at Test.proxy [as assert] (node_modules/sinon/lib/sinon/spy.js:82:26)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1419:10

  8) instrument supertest / superagent response can inspect responses with no bodies:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at responseHeaders (lib/instruments/supertest.js:57:22)
      at extractResponse (lib/instruments/supertest.js:43:14)
      at Test.exports.assert (lib/instruments/supertest.js:15:15)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:156:55)
      at Test.proxy [as assert] (node_modules/sinon/lib/sinon/spy.js:82:26)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1419:10
...
```
It is getting a bit late in the night for me and I have had a most awful couple of days wrestling with other package issues. But I would be absolutely delighted to lend a hand fixing this if it is within my capabilities.Anyway I see it is morning time in Australia, I look forward to talking to you.

**Comment by PatrickNausha** [comment]
I would love to help fix this as well. I see a similar errors when running with supertest 1.0.1.

Example:
```
2) app items gets items:
     Uncaught TypeError: Object.keys called on non-object
      at Function.keys (native)
      at net.js:1276:10
```

Very cool library @rprieto. Such a pragmatic way to create API documentation. I will be likely showing this library off to my company.

**Comment by vickvu** [comment]
this issue is fixed in this PR https://github.com/rprieto/supersamples/pull/27

**Comment by rprieto** [comment]
Thanks @ghinks for the very detailed report, and @vickvu for the fix! I'll merge this ASAP and publish a new version.

Glad to hear it's a useful project @PatrickNausha, please pass on any feedback if you have some :)

**Comment by rprieto** [comment]
Published `supersamples@1.0.0`, let me know if it works!

**Comment by ghinks** [comment]
Yes, it works. My considerable thanks to the team.
